### ISSUE TITLE
Make links to posts, communities or users on other lemmy instances federate the [whatever is being linked] instead of linking to that instance

### ISSUE THREAD

**Issue by Lilith-In-Starlight** [LemmyNet/lemmy-ui#233]
### Is your proposal related to a problem?

When clicking on a link to something that comes from another instance (which is very often since I use sopuli.xyz and most Things are on lemmy.ml), it links me to the link in that instance, which can be pretty frustrating if I wish to interact with it.

### Describe the solution you'd like

Making it so that when a link directs to something from another instance, instead of taking me to it, making it federate it to the current instance and open it there instead. Blocked instances would be exempt from this.

### Describe alternatives you've considered

The way I work around this is having an account on lemmy.ml and using it when a post hasn't been federated to sopuli.xyz, I just upvote it so that it gets to sopuli. For communities or users I just search them in the !name@instance format.

**Comment by dessalines** [comment]
I'm confused, the link should be going to your local post url. Could you screenshot what link you're clicking?

**Comment by Lilith-In-Starlight** [comment]
It should? It's never been the case for me.

I have made this example in a test community I found. Clicking on the link in the comment I've made takes you to Lemmy. What I'm requesting is to somehow make it take you to that post but locally. https://sopuli.xyz/post/3806 (the link in my comment goes to https://lemmy.ml/post/59022)

**Comment by dessalines** [comment]
I don't think its a good idea to manipulate comment text like that. If they link to something, that's an explicit link to another instance. Otherwise they can link to their local version of it. There's no guarantee that link exists on other instances either.

**Comment by Nutomic** [comment]
I disagree, I think it makes sense that Lemmy links should be opened directly in your own instance. That way you can directly vote, comment etc which is impossible otherwise. Users can link to their local version, but federated users would still end up on a different instance when clicking the link.

Now how to implement that I'm not sure. One option would be to rewrite links in post/comment text, if we can fetch the link target as apub object.
